extensions/sd_api_pictures/script.py

```diff
@@ -9,52 +10,92 @@
 import requests
 import torch
 from PIL import Image
+from datetime import date
 
 torch._C._jit_set_profiling_mode(False)
 
 # parameters which can be customized in settings.json of webui  
 params = {
-    'enable_SD_api': False,
-    'address': 'http://127.0.0.1:7860',
+    'address': '127.0.0.1:7860',
+    'mode': 0, # modes of operation: 0 (Manual only), 1 (Immersive/Interactive - looks for words to trigger), 2 (Picturebook Adventure - Always on)
+    'manage_VRAM': False,
     'save_img': False,
-    'SD_model': 'NeverEndingDream', # not really used right now
-    'prompt_prefix': '(Masterpiece:1.1), (solo:1.3), detailed, intricate, colorful',
+    'SD_model': 'NeverEndingDream', # not used right now
+    'prompt_prefix': '(Masterpiece:1.1), detailed, intricate, colorful',
     'negative_prompt': '(worst quality, low quality:1.3)',
-    'side_length': 512,
+    'width': 512,
+    'height': 512,
     'restore_faces': False
 }
 
+def update_params(): # somewhy the default extension params dict is not changed by settings.json, thus update it forcefully
```

Comment:
This sounds like it'll be worth looking into and fixing

Reply:
I've looked into it and it's being done, but just a bit too late. 

If something is happening right as the extension is loading -- like, for example, setting initial UI state (names of the buttons), or [checking if local cache exists (for Silero-TTS) and switching to it without connecting to Net](https://github.com/oobabooga/text-generation-webui/pull/628/files#diff-428f3d2a0bf16166af01f47be473a94fa202e4844e2ea38547d5d2742d16cf20R62) it's already too late.

Reply:
The fix is a subject for a separate PR but I think what could be done is to have an explicit `def on_load():` or similar function, such that the order of calls is: (1) file init, (2) settings load, (3) `on_load` called, and thus any init behavior that needs settings can simply be put inside the `on_load` instead of in the file init.
As a bonus, if it's ordered such that _all file inits_ for all extensions happen, _then_ all `on_load`, in separate groups, it would enable the ability for extensions to interact with each other.